src/sql/parts/objectExplorer/common/objectExplorerService.ts

```diff
@@ -320,41 +356,95 @@ export class ObjectExplorerService implements IObjectExplorerService {
+					let allProviders: sqlops.ObjectExplorerProviderBase[] = [provider];
+
+					let nodeProviders = this._nodeProviders[providerId];
+					if (nodeProviders) {
+						nodeProviders = nodeProviders.sort((a, b) => a.group.toLowerCase().localeCompare(b.group.toLowerCase()));
+						allProviders.push(...nodeProviders);
 					}
+
+					self._sessions[session.sessionId].nodes[nodePath].expandEmitter.event((expandResult) => {
+						if (expandResult) {
+							if (expandResult.providerId) {
+								resultMap.set(expandResult.providerId, expandResult);
+							} else {
+								reject(expandResult.errorMessage ? expandResult.errorMessage : 'ProviderId is undefined in expandResult');
+							}
+						}
+						else {
+							reject(expandResult ? expandResult.errorMessage : undefined);
+						}
+
+						// When get all responses from all providers, merge results
+						if (resultMap.size === allProviders.length) {
+							resolve(self.mergeResults(allProviders, resultMap));
+
+							// Have to delete it after get all reponses otherwise couldn't find session for not the first response
+							if (newRequest) {
+								delete self._sessions[session.sessionId].nodes[nodePath];
+							}
+						}
+					});
 					if (newRequest) {
-						delete self._sessions[session.sessionId].nodes[nodePath];
+						allProviders.forEach(provider => {
+							TelemetryUtils.addTelemetry(this._telemetryService, TelemetryKeys.ObjectExplorerExpand, { refresh: 0, provider: providerId });
+							self.callExpandOrRefreshFromProvider(provider, {
+								sessionId: session.sessionId,
+								nodePath: nodePath
+							}, refresh).then(isExpanding => {
+								if (!isExpanding) {
+									// The provider stated it's not going to expand the node, therefore do not need to track when merging results
+									let emptyResult: sqlops.ObjectExplorerExpandInfo = {
+										errorMessage: undefined,
+										nodePath: nodePath,
+										nodes: [],
+										sessionId: session.sessionId
+									};
+									resultMap.set(provider.providerId, emptyResult);
+								}
+							}, error => {
+								reject(error);
+							});
+						});
 					}
-				}));
-				if (newRequest) {
-					self.callExpandOrRefreshFromProvider(provider, {
-						sessionId: session.sessionId,
-						nodePath: nodePath
-					}, refresh).then(result => {
-					}, error => {
-						reject(error);
-					});
 				}
 			} else {
 				reject(`session cannot find to expand node. id: ${session.sessionId} nodePath: ${nodePath}`);
 			}
 		});
 	}
 
+	private mergeResults(allProviders: sqlops.ObjectExplorerProviderBase[], resultMap: Map<string, sqlops.ObjectExplorerExpandInfo>): sqlops.ObjectExplorerExpandInfo {
+		let finalResult: sqlops.ObjectExplorerExpandInfo;
+		let allNodes: sqlops.NodeInfo[] = [];
+		for (let provider of allProviders) {
+			if (resultMap.has(provider.providerId)) {
+				let result = resultMap.get(provider.providerId);
+				if (result && !result.errorMessage) {
```

Comment:
OK so we're officially losing any error messages. This needs solving - chat with Matt for ideas. Worst case any error means we show failure for all. Ideally we have a state where we can show what nodes _were_ returned successfully, but if any provider returns an error we show error glyph for parent node and have way to see all error messages.
Example:
- If 1 provider has error message, use that
- if 2 or more have messages, use string `1 or more errors occurred: {0}` and then append all the error messages as newlines?